File: examples/python/clean_api_python/domain/repositories/stock_movement_repository.py
from abc import ABC, abstractmethod
from typing import List, Optional, Tuple

from ..entities.stock_movement import StockMovement


class IStockMovementRepository(ABC):
    @abstractmethod
    def save(self, movement: StockMovement) -> StockMovement: ...

    @abstractmethod
    def find_by_product_id(
        self, product_id: Optional[str], page: int = 1, page_size: int = 10
    ) -> Tuple[List[StockMovement], int]: ...

File: examples/go/clean-api-go/domain/repositories/stock_movement_repository.go
package repositories

import (
	"context"

	"github.com/freakjazz/clean-api-go/domain/entities"
)

type IStockMovementRepository interface {
	Save(ctx context.Context, movement *entities.StockMovement) (*entities.StockMovement, error)
	FindByProductID(ctx context.Context, productID string, page, pageSize int) ([]*entities.StockMovement, int, error)
}
